pkg/gadgets/run/tracer/tracer.go

```diff
@@ -176,6 +180,21 @@ func (t *Tracer) installTracer() error {
 		}
 	}
 
+	// Only create socket enricher if this is used by the tracer
+	for _, m := range t.spec.Maps {
+		if m.Name == networktracer.SocketsMapName {
```

Comment:
At the moment, `networktracer.SocketsMapName` is defined as follows:
```
SocketsMapName = "sockets"
```

What do you think about renaming it to "gadget_sockets" so that it is namespaced correctly for third-party gadget authors?

Reply:
Sure, I think we must rename that to avoid collisions.

Reply:
I renamed it with the "ig_" prefix like some other maps and progs.

I also moved the constant from the `networktracer` package to the `socketenricher` package where it semantically belongs.

Reply:
> I renamed it with the "ig_" prefix like some other maps and progs.

I'm trying to use `gadget_` everywhere as in https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/gadgets/common/mntns_filter.h for the API. I think we're using `ig_` in the programs because the length restriction. However I don't have a preference in any of them, but I'd prefer everything aligned.

Reply:
Ok, I renamed to gadget_sockets.

I will merge once the CI finishes.